src/biogeochem/DustEmisLeung2023.F90

```diff
@@ -0,0 +1,874 @@
+            if (vai_Okin(p) > 1_r8) then
+               vai_Okin(p)  = 1_r8   ! setting LAI = 1 to be a max value (since K_length goes to negative when LAI>1) 
+            end if
+
+
+            ! calculate Okin's shear stress ratio (SSR, which is vegetation drag partition factor) using Pierre's equation
+            K_length = 2_r8 * (1_r8/vai_Okin(p) - 1_r8)   ! Here LAI has to be non-zero to avoid blowup, and < 1 to avoid -ve K_length. See this equation in Leung et al. (2023). This line is Okin's formulation
+            ssr(p) = (K_length+f_0*c_e)/(K_length+c_e) ! see this equation in Caroline Pierre et al. (2014) or Leung et al. (2023). This line is Pierre's formulation.
+
+            ! calculation of the hybrid/total drag partition effect considering both rock and vegetation drag partitioning using LUH2 bare and veg fractions within a grid
+            if (lun%itype(l) == istsoil .or. lun%itype(l) == istcrop) then
+               if (patch%itype(p) == noveg) then ! if bare, uses rock drag partition factor
+                  if (dpfct_rock(p) /= dpfct_rock(p)) then ! dmleung added 24 May 2024: dpfct_rock(p) could be NaN; CLM could run when DEBUG=FALSE in env_build.xml but dies when DEBUG=TRUE (usually when checking if wnd_frc_slt > wnd_frc_thr_slt_it and if numer/denom < 30._r8 below)
+                     !write(iulog,*) 'dpfct_rock(p) == NaN; dpfct_rock(p) = ', dpfct_rock(p)
+                     frc_thr_rgh_fct = 0.001_r8 ! Set drag partition effect to be a very small value (or zero) such that there is no emission whenever dpfct_rock(p) = NaN; dmleung 24 May 2024
+                  else
+                     !write(iulog,*) 'dpfct_rock(p) = ', dpfct_rock(p)
+                     frc_thr_rgh_fct = dpfct_rock(p)
+                  end if
+               !frc_thr_rgh_fct = dpfct_rock(p) ! This should be the original code when dpfct_rock(p) has values everywhere
+               else   ! if vegetation, uses vegetation drag partition factor
+                  frc_thr_rgh_fct = ssr(p)
+               end if
+            else
+               frc_thr_rgh_fct = 1.0_r8
+            end if
+
+            wnd_frc_slt = fv(p) * frc_thr_rgh_fct   ! wnd_frc_slt is the drag-parition-modified wind speed and will be used in the dust emission equation below
+
+            frc_thr_rghn_fct(p) = frc_thr_rgh_fct   ! save and output hybrid drag partition factor
+
+         else    ! for lnd_frc_mbl=0, do not change friction velocity and assume drag partition factor = 0
+            wnd_frc_slt = fv(p) * frc_thr_rghn_fct(p) ! The value here is not important since once lnd_frc_mbl(p) <= 0.0_r8 there will be no emission. dmleung added 5 Jul 2024
+            frc_thr_rghn_fct(p) = 0.0_r8            ! When LAI > 1, the drag partition effect is zero. dmleung 16 Feb 2024.
```

Comment:
This is a comment, but another point where the 1 should be vai_mbl_thr

Reply:
Right, yes, so go back to line 645, this is the else if lnd_frc_mbl=0 or tlai_lu(l)>vai_mbl_thr. Thanks